jetty-core/jetty-http2/jetty-http2-client-transport/src/main/java/org/eclipse/jetty/http2/client/transport/internal/ClientHTTP2StreamEndPoint.java

```diff
@@ -34,26 +38,51 @@ public ClientHTTP2StreamEndPoint(HTTP2Stream stream)
     }
 
     @Override
-    public void onDataAvailable()
+    public Runnable onDataAvailable()
     {
-        processDataAvailable();
+        // The InvocationType may change depending on the read callback.
+        return new Invocable.ReadyTask(getInvocationType(), this::processDataAvailable);
```

Comment:
Can the `ReadyTask` be created once an reused?

Reply:
As the comment says, the read callback may have a different invocation type, depending on the protocol on top.